WebRoot/js/zimbraAdmin/resource/view/ZaNewResourceXWizard.js

```diff
@@ -510,14 +510,17 @@ ZaNewResourceXWizard.myXFormModifier = function(xFormObject) {
 						{type:_ZAWIZGROUP_, colSizes:["200px","300px"],
 							items:[
 								{type:_GROUP_, numCols:3, nowrap:true, width:200, msgName:ZaMsg.NAD_LocationDisplayName,label:ZaMsg.NAD_LocationDisplayName, labelLocation:_LEFT_, 
+									visibilityChecks:[[ZaItem.hasWritePermission,ZaResource.A_locationDisplayName]],
 									items: [
 										{ref:ZaResource.A_locationDisplayName, type:_TEXTFIELD_, 
 											label:null,	width:defaultWidth,
-											enableDisableChecks:[ZaNewResourceXWizard.isAutoDisplayname],
+											enableDisableChecks:[ZaNewResourceXWizard.isAutoDisplayname, ZaItem.hasWritePermission],
                                                                                         enableDisableChangeEventSources:[ZaResource.A2_autoLocationName],bmolsnr:true
 										},
 										{ref:ZaResource.A2_autoLocationName, type:_WIZ_CHECKBOX_, msgName:ZaMsg.NAD_Auto,label:ZaMsg.NAD_Auto,labelLocation:_RIGHT_,trueValue:"TRUE", falseValue:"FALSE",
-											elementChanged: ZaResource.setAutoLocationName
+											elementChanged: ZaResource.setAutoLocationName,
+											enableDisableChecks:[[ZaItem.hasWritePermission,ZaResource.A_locationDisplayName]],
+											visibilityChecks:[[ZaItem.hasWritePermission,ZaResource.A_locationDisplayName]]
```

Comment:
shouldn't we need to check for read permission for visibility?

Reply:
On New account/dl/resource wizard, fields that an admin does not have write permission are hidden rather than disabled.
In this case, "Location Display Name" field is hidden if the admin doesn't have write permission. Then "auto" checkbox should also be hidden.
So visibilityChecks checks write permission here.

Example of other fields:

    ZaNewAccountXWizard.js
        visibilityChecks:[[ZaItem.hasWritePermission,ZaAccount.A_COSId]],
        visibilityChecks:[[ZaItem.hasWritePermission,ZaResource.A_accountStatus]],